File: trading-chats-frontend/src/api/systemConfig.ts
import { http } from './http'
import type { ApiResponse, SystemConfig } from './types'
import { unwrap } from './unwrap'

export type SaveSystemBasicConfigRequest = Pick<SystemConfig, 'system_title' | 'system_logo'>
export type SaveSystemParametersRequest = Pick<SystemConfig, 'parameters'>

export async function getSystemConfig(): Promise<SystemConfig> {
  const res = await http.get<ApiResponse<SystemConfig>>('/system-config')
  return unwrap(res.data)
}

export async function saveSystemBasicConfig(config: SaveSystemBasicConfigRequest): Promise<void> {
  const res = await http.put<ApiResponse<void>>('/system-config/basic', config)
  unwrap(res.data)
}

export async function saveSystemParameters(config: SaveSystemParametersRequest): Promise<void> {
  const res = await http.put<ApiResponse<void>>('/system-config/parameters', config)
  unwrap(res.data)
}

File: trading-chats-backend/internal/models/system_config.go
package models

import "time"

const GlobalSystemConfigID = "global_config"

type SystemConfig struct {
	ID          string            `bson:"_id,omitempty" json:"id"`
	SystemTitle string            `bson:"system_title" json:"system_title"`
	SystemLogo  string            `bson:"system_logo" json:"system_logo"`
	Parameters  map[string]string `bson:"parameters" json:"parameters"`
	UpdatedAt   time.Time         `bson:"updated_at" json:"updated_at"`
}

type SaveSystemBasicConfigRequest struct {
	SystemTitle string `json:"system_title"`
	SystemLogo  string `json:"system_logo"`
}

type SaveSystemParametersRequest struct {
	Parameters map[string]string `json:"parameters"`
}
